packages/react/src/Details/Details.tsx

```diff
@@ -14,7 +16,55 @@ const Details = styled.details<SxProp>`
   ${sx};
 `
 
-Details.displayName = 'Details'
+export const StyledSummary = styled.summary<SxProp>`
+  ${sx};
+`
+
+const Root = React.forwardRef<HTMLDetailsElement, ComponentProps<typeof StyledDetails>>(
+  ({children, ...props}: ComponentProps<typeof StyledDetails>, ref) => {
+    const hasSummary = React.Children.toArray(children).some(child => {
+      return (
+        React.isValidElement(child) &&
+        (child.type === Details.Summary || child.type === 'summary' || child.props?.as === 'summary')
+      )
+    })
```

Comment:
This kind of pattern with children is tricky because it will end up breaking composition 🤔 Is there a good way to provide this otherwise or not really?

Specifically, if I ever wrap `Details.Summary` in my own component (like `IssueSummary` that renders `Details.Summary`) this would not work anymore even though it renders the correct thing.

Reply:
Yeah that's true. Wondering if we want to rethink the suggestion to add a default then: 
> We recommend extending the component to provide a ready-made slot/property that will output a `<summary>` by default. 

Not sure how we could detect presence of a summary within the `Details` otherwise 🤔 

I do see we have some precedence for this type of behavior with slots: https://github.com/primer/react/blob/main/packages/react/src/experimental/SelectPanel2/SelectPanel.tsx#L314, https://github.com/primer/react/blob/main/packages/react/src/ActionList/Group.tsx#L86 

Any thoughts? 👀